library/alloc/src/vec/mod.rs

```diff
@@ -1231,6 +1237,16 @@ impl<T, A: Allocator> Vec<T, A> {
     /// }
     /// ```
     ///
+    /// The validity guarantee works out this way:
```

Comment:
I wouldn't call it "validity guarantee", that sounds related to validity invariants. Maybe "aliasing guarantee"?